selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java

```diff
@@ -655,14 +656,23 @@ public void waitRemoveItemsByPath(String pathToItem) {
    * @param path item's path in format: "Test/src/pom.xml".
    */
   public void openContextMenuByPathSelectedItem(String path) {
-    waitItem(path);
-    waitAndSelectItem(path);
-    waitItemIsSelected(path);
-
-    Actions action = actionsFactory.createAction(seleniumWebDriver);
-    action.moveToElement(waitAndGetItem(path)).contextClick().perform();
+    for (int i = 1; ; i++) {
```

Comment:
Could you explain what the problem was here and how the current code fixes it?

Reply:
Selecting item and invoking context menu are two separate operations. 
 If some time retakes focus then context menu is invoked at wrong item. It might happen because project explorer updates asynchronously and new appeared items retake focus.
So, there are two tries to invoke context menu at correct item